### PR TITLE
Fix accidental removal of bzip2 on RHEL

### PR DESCRIPTION
**Pull request by scotthain**
@chef-cookbooks/engineering-services @chef-cookbooks/omnibus-maintainers

### CODE REVIEW

File: recipes/_compile.rb [hunk 1]
```diff
@@ -78,6 +78,9 @@
   #   https://github.com/CentOS/sig-cloud-instance-images/issues/4
   #
   package 'tar'
+  # Don't introduce a bug on RHEL by not having bzip2
+  # this was fixed previously in omnibus v2.4.12
+  package 'bzip2'
```
Comment (yzl) [inline recipes/_compile.rb]: Sorry, I seem to be writing a lot of comments about comments lately, but:

* I'm missing context on why this comment is needed.  Is removal of the `bzip2` something that keeps biting us?
* Assuming that it is needed, I'd put that context in the comment.

Comment (scotthain) [inline recipes/_compile.rb]: @yzl Technically one could say that I introduced this bug back again in 4.0.0 by pruning the code that ensured bzip2 was installed. I think removing the comment probably would be best in this case? I was being overly explanatory I think...

Comment (yzl) [inline recipes/_compile.rb]: @scotthain Removing the comment seems reasonable.  If we break ourselves again by removing the install, we can add it back with a comment :-)

**Comment by schisamo** [comment]
@scotthain I'm having trouble locating the commit where this was removed.

**Comment by scotthain** [comment]
@schisamo it was here when the ruby compile style changed: https://github.com/chef-cookbooks/omnibus/commit/a49500fd39642fbc44f4f9e9deb34b45342157be

The reason this is here in general is because on ppc64 RHEL7, when uncompressing fakeroot, it fails.

**Comment by schisamo** [comment]
:+1:

**Pull request merged**